fix: Persist stale cells on session resume

wasimsandhu opened the pull request:
## 📝 Summary

Fixes #1444, an edge case bug that caused cells to lose "stale" status on session resume.

Important to note that this bug was only for uninstantiated cells, or effectively, the first session resume. Cells that had been run and later became stale were already persisted on subsequent resumes.

## 🔍 Description of Changes

- In `createCellRuntimeState()`, pass `staleInputs: true` if `lastExecutionTime == null`, which will only be null for an uninstantiated cell.
- Update cell status tooltip with more relevant "This cell has not yet been run" hint for uninstantiated cells.'

Uninstantiated cells after session resume:
<img width="1376" alt="Screenshot 2024-06-16 at 1 06 57 PM" src="https://github.com/marimo-team/marimo/assets/7220175/f21fa951-33cd-4e30-849b-cde84901eb96">
Stale cells after session resume:
<img width="1448" alt="Screenshot 2024-06-16 at 4 53 13 PM" src="https://github.com/marimo-team/marimo/assets/7220175/d3d46edf-5cfb-4084-9425-d0cd972de6b8">

## 📋 Checklist

- [x] I have read the [contributor guidelines](../CONTRIBUTING.md).
- [x] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://discord.gg/JE7nhX6mD8), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [ ] I have added tests for the changes made.
- [x] I have run the code and verified that it works as expected.

## 📜 Reviewers

@mscolnick

mscolnick commented:
awesome! nice fix and improved messaging!

akshayka commented:
@wasimsandhu @mscolnick I think there may be a regression with this PR, with cells incorrectly marked as stale on startup.

- I have autorun on startup enabled.
- I ran `marimo tutorial intro`
- The notebook correctly ran, but everything is incorrectly marked as stale:

<img width="949" alt="image" src="https://github.com/marimo-team/marimo/assets/1994308/93fac592-86ee-4826-b090-3209d6da7230">

wasimsandhu commented:
After testing the 4 runtime combinations, I removed a line of code that I thought was unnecessary and only tested with autorun on startup disabled. Sorry about that! Fixed in #1626.
